utilities/transactions/write_prepared_txn_db.h

```diff
@@ -124,6 +124,9 @@ class WritePreparedTxnDB : public PessimisticTransactionDB {
                       "IsInSnapshot %" PRIu64 " in %" PRIu64
                       " min_uncommitted %" PRIu64,
                       prep_seq, snapshot_seq, min_uncommitted);
+    if (snap_released != nullptr) {
+      *snap_released = false;
```

Comment:
This statement would be call on every single IsInSnapshot call. How about the caller have the responsibility to initialize it to false? In this way only the ones who use it pay the cost.